EIPS/eip-5750.md

```diff
@@ -8,103 +8,91 @@ status: Review
 2. Having the bytes specified as the last parameter makes this EIP compatible with the calldata layout of solidity.
+3. Requiring EIP-165 interface identifiers to include the overloads does not break backward compatibility.
 
 ## Backwards Compatibility
 
-Many of existing EIPs already have compliant methods and all contracts compliant with those EIPs are already compliant with this EIP.
+Many existing EIPs already have compliant methods as part of their specification. All contracts compliant with those EIPs are either fully or partially compliant with this EIP.
 
 Here is an incomplete list:
 
-1. In [EIP-721](./eip-721.md) the following methods are already compliant:
-
-- `function safeTransferFrom(address _from, address _to, uint256 _tokenId, bytes data) external payable;` is already compliant
-
-2. In [EIP-1155](./eip-1155.md) the following methods are already compliant
+* In [EIP-721](./eip-721.md), the following method is already compliant:
+  * `function safeTransferFrom(address _from, address _to, uint256 _tokenId, bytes data) external payable;` is already compliant
+* In [EIP-1155](./eip-1155.md), the following methods are already compliant
+  * `function safeTransferFrom(address _from, address _to, uint256 _id, uint256 _value, bytes calldata _data) external;`
+  * `function safeBatchTransferFrom(address _from, address _to, uint256[] calldata _ids, uint256[] calldata _values, bytes calldata _data) external;`
+* In [EIP-777](./eip-777.md), the following methods are already compliant
+  * `function burn(uint256 amount, bytes calldata data) external;`
+  * `function send(address to, uint256 amount, bytes calldata data) external;`
 
-- `function safeTransferFrom(address _from, address _to, uint256 _id, uint256 _value, bytes calldata _data) external;`
-- `function safeBatchTransferFrom(address _from, address _to, uint256[] calldata _ids, uint256[] calldata _values, bytes calldata _data) external;`
+However, not all functions that have a `bytes` as the last parameter are compliant. The following functions are not compliant without an overload since their last parameter is involved in functionality:
 
-3. In [EIP-777](./eip-777.md) the following methods are already compliant
-
-- `function burn(uint256 amount, bytes calldata data) external;`
-- `function send(address to, uint256 amount, bytes calldata data) external;`
-
-4. In [EIP-2535](./eip-2535.md) the following methods are already compliant
-
-```solidity
-function diamondCut(
-        FacetCut[] calldata _diamondCut,
-        address _init,
-        bytes calldata _calldata
-    ) external;
-```
-
-5. In [EIP-1271](./eip-1271.md) the following methods are already compliant:
-
-```solidity
-  function isValidSignature(
-    bytes32 _hash,
-    bytes memory _signature)
-    public
-    view
-    returns (bytes4 magicValue);
-```
+* In [EIP-2535](./eip-2535.md), the following methods is not compliant:
```

Comment:
I understand in your proposed changes to specification because EIP-2135 and EIP-1271 was using the last data for special purpose they are not considered compliant.

I am weakly leaning towards allowing them as compliant instead to allow maximum compatibility. We can chat more about this

Reply:
I think that they should not be compliant. The self-stated purpose of this EIP is to allow arbitrary future extensions. Thus, future extensions should not be limited by restrictions on existing parameters, as that would not allow arbitrary future extensions. Thus, my wording is preferable.

Reply:
Ok, let's get the wording in first and I will think a bit more on how to resolve the conflicting situation. i appreciate that you highlighted the potential ambiguity. Let's merge this commit in first